src/main/java/org/mskcc/cbio/oncokb/web/rest/CompanyResource.java

```diff
@@ -195,4 +201,35 @@ public ResponseEntity<Void> deleteCompany(@PathVariable Long id) {
         companyService.delete(id);
         return ResponseEntity.ok().build();
     }
+
+    @GetMapping("/companies/licenses-about-to-expire-pending-notification")
+    public List<Long> getLicensesAboutToExpirePendingNotification() {
+        log.debug("Request to get Company license is about to expire");
+        Instant now = Instant.now();
+        ArrayList<Long> companies = new ArrayList<>();
+        for (CompanyDTO company : companyService.findAll()) {
+           CompanyAdditionalInfoDTO additionalInfo = company.getAdditionalInfo();
+           if (additionalInfo != null) {
+               CompanyLicense license = additionalInfo.getLicense();
+               if (license != null) {
+                   CompanyTermination termination = license.getTermination();
+                   if (termination != null) {
```

Comment:
You can do Optional chaining to make it more readable. https://stackoverflow.com/a/35337260

Reply:
![image](https://github.com/user-attachments/assets/783e9793-b0b8-452b-aaa9-826b360bcd75)